pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,617 @@
+				},
+			},
+			Phase: v1.PodSucceeded,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "pod1",
+			Namespace:         "pod1_ns",
+			UID:               "pod4",
+			CreationTimestamp: metav1.Time{Time: refTime.Add(-5 * time.Minute)},
+			DeletionTimestamp: &metav1.Time{Time: refTime},
+		},
+	}
+	timestampMap = map[types.UID]*podLocalTimestamps{
+		"pod1": {CreationTimestamp: refTime},
+		"pod2": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
+		"pod4": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
+	}
+	node = &v1.Node{}
+)
+
+func Test_onPodUpdate(t *testing.T) {
+	newPod1 := &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "4.5.6.7",
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod1",
+			Namespace: "pod1_ns",
+			UID:       "pod1",
+		},
+	}
+	tests := []struct {
+		name        string
+		podStore    *PodStore
+		oldObj      interface{}
+		newObj      interface{}
+		expectedPod *v1.Pod
+	}{
+		{
+			name: "newObj is not Pod",
+			podStore: &PodStore{
+				pods: cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			newObj:      node,
+			expectedPod: pod1,
+		},
+		{
+			name: "Valid case",
+			podStore: &PodStore{
+				pods: cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			newObj:      newPod1,
+			expectedPod: newPod1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.podStore.pods.Add(pod1)
+			tt.podStore.onPodUpdate(tt.oldObj, tt.newObj)
+			assert.Equal(t, 1, len(tt.podStore.pods.List()))
+			assert.Equal(t, tt.expectedPod, tt.podStore.pods.List()[0].(*v1.Pod))
+		})
+	}
+}
+
+func Test_onPodCreate(t *testing.T) {
+	tests := []struct {
+		name                    string
+		podStore                *PodStore
+		obj                     interface{}
+		expectedTimestampMapLen int
+		expectedCreationTime    time.Time
+	}{
+		{
+			name: "object is not Pod",
+			podStore: &PodStore{
+				timestampMap: map[types.UID]*podLocalTimestamps{},
+				clock:        clock.NewFakeClock(refTime),
+				pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			obj:                     node,
+			expectedTimestampMapLen: 0,
+		},
+		{
+			name: "Valid case for Pending Pod",
+			obj:  pod1,
+			podStore: &PodStore{
+				timestampMap: map[types.UID]*podLocalTimestamps{},
+				clock:        clock.NewFakeClock(refTime),
+				pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			expectedTimestampMapLen: 1,
+			expectedCreationTime:    refTime,
+		},
+		{
+			name: "Valid case for Running Pod",
+			obj:  pod3,
+			podStore: &PodStore{
+				timestampMap: map[types.UID]*podLocalTimestamps{},
+				clock:        clock.NewFakeClock(refTime),
+				pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
```

Comment:
you seem to be repeating the same PodStore initialization for all 3 test cases. Instead you could do it in the body of the test (inside `t.Run`); you don't need to make it a field of the test case struct.

This comment also applies to other unit tests in this this file.

Reply:
Updated.